internal/docstore/driver/codec.go

```diff
@@ -369,6 +370,9 @@ func Decode(v reflect.Value, d Decoder) error {
 }
 
 func decode(v reflect.Value, d Decoder) error {
+	if !v.CanSet() {
+		panic(fmt.Sprintf("cannot set %+v", v))
```

Comment:
Maybe return an error instead of panicking. If we don't do this it will also panic when we try to set the value right?

Reply:
Right.

Done.